src/test/java/org/junitpioneer/jupiter/RetryingTestExtensionTests.java

```diff
@@ -220,6 +228,63 @@ void minSuccessLessThanOne_fails() {
 		assertThat(results).hasNumberOfDynamicallyRegisteredTests(0);
 	}
 
+	@Test
+	void suspendForLessThanZero_fails() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethod(RetryingTestTestCases.class, "suspendForLessThanZero");
+
+		assertThat(results).hasNumberOfDynamicallyRegisteredTests(0);
+	}
+
+	@Test
+	void failThreeTimesWithSuspend() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethod(RetryingTestTestCases.class, "failThreeTimesWithSuspend");
+
+		assertThat(results)
+				.hasNumberOfDynamicallyRegisteredTests(3)
+				.hasNumberOfAbortedTests(2)
+				.hasNumberOfFailedTests(1)
+				.hasNumberOfSucceededTests(0);
+
+		assertSuspendedForCloseTo(results, SUSPEND_FOR);
+	}
+
+	@Test
+	void failThreeTimesWithoutSuspend() {
+		ExecutionResults results = PioneerTestKit
+				.executeTestMethod(RetryingTestTestCases.class, "failThreeTimesWithoutSuspend");
+
+		assertThat(results)
+				.hasNumberOfDynamicallyRegisteredTests(3)
+				.hasNumberOfAbortedTests(2)
+				.hasNumberOfFailedTests(1)
+				.hasNumberOfSucceededTests(0);
+
+		assertSuspendedForCloseTo(results, 0);
+	}
+
+	private void assertSuspendedForCloseTo(ExecutionResults results, long closeTo) {
+		List<Execution> finishedExecutions = results.testEvents().executions().finished().list();
+		List<Execution> startedExecutions = results.testEvents().executions().started().list();
+
+		// Compare 'finished' execution timestamp with follow-up 'started' execution,
+		// skipping the 1st 'started' and final 'finished'.
+		// This duration should be close to 'suspendFor'.
+		for (int i = 0; i < finishedExecutions.size() - 1; i++) {
+			Execution finished = finishedExecutions.get(i);
+			Execution nextStarted = startedExecutions.get(i + 1);
+
+			Instant finishedAt = finished.getEndInstant();
+			Instant nextStartedAt = nextStarted.getStartInstant();
+
+			long suspendedFor = Duration.between(finishedAt, nextStartedAt).toMillis();
+
+			// We allow an error margin since testing timings is not an exact science.
```

Comment:
Indeed, already the second build failed because of that. 😬 I think it's ok to just assert that the actual suspension time is equal to or larger than the configured time. As long as we don't do additional math on the attribute (like `Thread.sleep(suspendInMs x 100)` 😁 ) that's perfectly fine.